engine/src/main/java/org/terasology/network/internal/ClientConnectionHandler.java

```diff
@@ -98,6 +101,15 @@ public void messageReceived(ChannelHandlerContext ctx, MessageEvent e) {
 
         // Handle message
         NetData.NetMessage message = (NetData.NetMessage) e.getMessage();
+        String errorMessage = message.getServerInfo().getErrorMessage();
+        if (errorMessage != null && !errorMessage.isEmpty()) {
+            synchronized (joinStatus) {
+                joinStatus.setErrorMessage(errorMessage);
```

Comment:
About localization: you may send the id of a string to take from the translation system instead of the human-readable English-only message. Then here you would need to access an instance of `TranslationSystem`, probably via `context`, to actually load the translated string.

However it may not be trivial to get access to a `Context` from `ClientConnectionHandler`... you may want to evaluate if it's worth it, in my opinion localizing such a short string would be nice if it's simple but not much of a big deal if it takes effort - that would be the job for another patch/PR (as you notice there are some other strings around which are not localized)

Reply:
It would be nice to have translations here, but it's not trivial like you mentioned, and I think it belongs in a different PR.